hack/util.sh

```diff
@@ -82,12 +82,16 @@ capz::util::generate_ssh_key() {
     AZURE_SSH_PUBLIC_KEY_FILE=${AZURE_SSH_PUBLIC_KEY_FILE:-""}
     if [ -z "${AZURE_SSH_PUBLIC_KEY_FILE}" ]; then
         echo "generating sshkey for e2e"
-        AZURE_SSH_KEY=.sshkey
+        AZURE_SSH_KEY="${PWD}"/.sshkey
+        # This is required if e2e.test is run with --provider=skeleton
```

Comment:
what is `--provider=skeleton`? can you please provide more context?

Reply:
That is the option passed to the `e2e.test` binary when running Windows e2e test against capz clusters. Ref: [kubernetes-sigs/windows-testing/capz/run-capz-e2e.sh](https://github.com/kubernetes-sigs/windows-testing/blob/be73dd57a551be4f9527e7a3a3e6491e86cae6d2/capz/run-capz-e2e.sh#L300). From what I understand this is being done to ensure that the e2e tests don't create any other Azure resources. cc: @jsturtevant

Reply:
could we add that context / link in the comment?

Reply:
Done. I added some extra context for a more complete explanation.

Reply:
That's correct, I beleive the conformance tests with capi framework use e2.test with the skeleton option